kratos/utilities/nurbs_utilities/projection_nurbs_geometry_utilities.h

```diff
@@ -0,0 +1,197 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Thomas Oberbichler
+//                   Tobias Teschemacher
+//
+
+#if !defined(KRATOS_PROJECTION_NURBS_GEOMETRY_UTILITIES_H_INCLUDED)
+#define KRATOS_PROJECTION_NURBS_GEOMETRY_UTILITIES_H_INCLUDED
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "containers/array_1d.h"
+
+namespace Kratos
+{
+    template<int TDimension, class TPointType> class NurbsCurveGeometry;
+    template<int TDimension, class TPointType> class NurbsSurfaceGeometry;
+    namespace ProjectionNurbsGeometryUtilities
+    {
+        typedef array_1d<double, 3> CoordinatesArrayType;
+
+        /*
+        * @brief Returns the projection of a point onto a Nurbs curve
+        *        geometry using the Newton-Rapshon iterative method
+        * @param rParameter Intial guess for the Newton-Rapshon algorithm
+        *        overwritten by the local coordinates of the projected point onto
+        *        the Nurbs curve geometry
+        * @param rPoint The point to be projected onto the Nurbs curve geometry
+        *        This is overwritten by the Cartesian coordinates of the projected
+        *        point in case the projection is successful 
+        * @param rResult The projection onto the Nurbs curve geometry
+        * @param rNurbsCurve The Nurbs curve geometry onto which the point is 
+        *        to be projected
+        * @param MaxIterations Maximum number of iterations for the Newton-Rapshon 
+        *        algorithm
+        * @param ModelTolerance Tolerance of the CAD model
+        * @param Accuracy Accuracy for the the Newton-Rapshon algorithm
+        */
+        template <int TDimension, class TPointType>
+        bool NewtonRaphsonCurve(
+            CoordinatesArrayType& rParameter,
+            const CoordinatesArrayType& rPointGlobal,
+            CoordinatesArrayType& rResultLocal,
+            const NurbsCurveGeometry<TDimension, TPointType>& rNurbsCurve,
+            const int MaxIterations = 20,
+            const double Accuracy = 1e-6,
+            const double ModelTolerance = 1e-6)
```

Comment:
We thought it would be better to withdraw the model tolerance including the breaking criteria because it is actually not part of the algorithm. Could you neglect it?